### DISCUSSION TITLE
[0.7] Ways of creating a non-Send Resource

### DISCUSSION THREAD

**Discussion by minijackson** [leptos-rs/leptos#3312] [Q&A] [answered]
Hey all! I'm trying to upgrade my Leptos project from 0.6 -> 0.7, and one thing I'm stuck on is that the Resource `fetcher` argument now has a `Send` bound. My fetcher code was indirectly using `reqwest::Response` which is `!Send` for WASM. What can I do?

**Comment by gbj** [comment, marked as answer]
The answer here depends slightly on which kind of platform you're using.

If you're using CSR, then the serialization/server-client part of Resource isn't necessary and you can use [LocalResource](https://docs.rs/leptos/0.7.0/leptos/prelude/struct.LocalResource.html), which doesn't have a `Send` bound.

If you're using SSR + hydration, then the problem is specifically that `reqwest` uses types that are `Send` on the server (and show up in docs as `Send`, which is confusing!) but are `!Send` in the WASM build. You might try taking a look at how I set this up in the [`hackernews` example](https://github.com/leptos-rs/leptos/blob/main/examples/hackernews/src/api.rs#L12-L58): essentially using `SendWrapper` on the client.

The added `Send` bound here is an unfortunate trade-off of allowing the reactive system and server rendering to support multithreading.

**Reply by minijackson** [reply]
Thanks a lot! I managed to get it working using LocalResource, since I'm only using that resource on the client side.
